app/views/fields/belongs_to/_show.html.erb

```diff
@@ -16,8 +16,12 @@ By default, the relationship is rendered as a link to the associated object.
 %>
 
 <% if field.data %>
-  <%= link_to(
-    field.display_associated_resource,
-    [Administrate::NAMESPACE, field.data],
-  ) %>
+  <% begin %>
+    <%= link_to(
+      field.display_associated_resource,
+      [Administrate::NAMESPACE, field.data],
+    ) %>
+  <% rescue %>
```

Comment:
I try to avoid using `rescue` without specifying a specific type of error - there's a good chance it can rescue from unrelated errors.

How about defining a `link_if_route_exists` helper?

```ruby
    def link_if_route_exists(text, *routing_args)
      link_to(text, *routing_args)
    rescue NoMethodError
      text
    end
```